Upstream-Status: Inappropriate [Backport]
From abb5bc5b7190b68909f9dd14386aa2c430bacd9a Mon Sep 17 00:00:00 2001
From: nickc <nickc@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Wed, 27 Apr 2011 14:55:28 +0000
Subject: [PATCH 189/200] 	* gcc.dg/20020312-2.c: Add definition for RX.

	* lib/target-supports.exp (add_options_for_ieee): Use -mnofpu
	option with RX targets.

	* gcc.target/rx/pack.c: Replace C++ style // comments with C style
	/* */ comments.


git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_6-branch@173033 138bc75d-0d04-0410-961f-82ee72b054a4

index bf8225a..0689636 100644
--- a/gcc/testsuite/gcc.dg/20020312-2.c
+++ b/gcc/testsuite/gcc.dg/20020312-2.c
@@ -58,6 +58,8 @@ extern void abort (void);
 # else
 #  define PIC_REG  "30"
 # endif
+#elif defined(__RX__)
+/* No pic register.  */
 #elif defined(__s390__)
 # define PIC_REG  "12"
 #elif defined(__sparc__)
diff --git a/gcc/testsuite/gcc.target/rx/pack.c b/gcc/testsuite/gcc.target/rx/pack.c
index 97c44f0..ce13b50 100644
--- a/gcc/testsuite/gcc.target/rx/pack.c
+++ b/gcc/testsuite/gcc.target/rx/pack.c
@@ -4,10 +4,10 @@ typedef unsigned short	INT16U;
 
 typedef struct tst_2
 {
-  INT16U	f0;	// [+0]
-  INT16U *	f1;	// [+2]
-  INT16U	f2;	// [+6]
-  INT16U *	f3;	// [+8]
+  INT16U	f0;	/* [+0] */
+  INT16U *	f1;	/* [+2] */
+  INT16U	f2;	/* [+6] */
+  INT16U *	f3;	/* [+8] */
 } __attribute__ ((__packed__)) t2;
 
 #include <stddef.h>
diff --git a/gcc/testsuite/lib/target-supports.exp b/gcc/testsuite/lib/target-supports.exp
index 36cab54..d2c6fff 100644
--- a/gcc/testsuite/lib/target-supports.exp
+++ b/gcc/testsuite/lib/target-supports.exp
@@ -3556,6 +3556,9 @@ proc add_options_for_ieee { flags } {
          || [istarget "sh*-*-*"] } {
        return "$flags -mieee"
     }
+    if { [istarget "rx-*-*"] } {
+       return "$flags -mnofpu"
+    }
     return $flags
 }
 
-- 
1.7.0.4

